manifests/init.pp

```diff
@@ -210,68 +232,100 @@
-  $snmp_config                  = $snmp::params::snmp_config,
-  Boolean $autoupgrade          = $snmp::params::autoupgrade,
-  $package_name                 = $snmp::params::package_name,
-  $snmptrapd_package_name       = $snmp::params::snmptrapd_package_name,
-  $snmpd_options                = $snmp::params::snmpd_options,
-  $service_config_perms         = $snmp::params::service_config_perms,
-  $service_config_dir_group     = $snmp::params::service_config_dir_group,
-  Stdlib::Ensure::Service $service_ensure = $snmp::params::service_ensure,
-  $service_name                 = $snmp::params::service_name,
-  Boolean $service_enable       = $snmp::params::service_enable,
-  Boolean $service_hasstatus    = $snmp::params::service_hasstatus,
-  Boolean $service_hasrestart   = $snmp::params::service_hasrestart,
-  $snmptrapd_options            = $snmp::params::snmptrapd_options,
-  Stdlib::Ensure::Service $trap_service_ensure = $snmp::params::trap_service_ensure,
-  $trap_service_name            = $snmp::params::trap_service_name,
-  $trap_service_enable          = $snmp::params::trap_service_enable,
-  $trap_service_hasstatus       = $snmp::params::trap_service_hasstatus,
-  $trap_service_hasrestart      = $snmp::params::trap_service_hasrestart,
-  String[1] $template_snmpd_conf          = $snmp::params::template_snmpd_conf,
-  String[1] $template_snmpd_sysconfig     = $snmp::params::template_snmpd_sysconfig,
-  String[1] $template_snmptrapd           = $snmp::params::template_snmptrapd,
-  String[1] $template_snmptrapd_sysconfig = $snmp::params::template_snmptrapd_sysconfig,
-  Boolean $openmanage_enable    = $snmp::params::openmanage_enable,
-  Boolean $master               = $snmp::params::master,
-  $agentx_perms                 = $snmp::params::agentx_perms,
-  $agentx_ping_interval         = $snmp::params::agentx_ping_interval,
-  $agentx_socket                = $snmp::params::agentx_socket,
-  Integer[0] $agentx_timeout    = $snmp::params::agentx_timeout,
-  Integer[0] $agentx_retries    = $snmp::params::agentx_retries,
-  Boolean $snmpv2_enable        = $snmp::params::snmpv2_enable,
-) inherits snmp::params {
+  Enum['present','absent']                                        $ensure        = 'present',
+  Array[String[1]]                                                $agentaddress  = [ 'udp:127.0.0.1:161', 'udp6:[::1]:161' ],
+  Array[String[1]]                                                $snmptrapdaddr = [ 'udp:127.0.0.1:162', 'udp6:[::1]:162' ],
+  Variant[Undef, String[1], Array[String[1]]]                     $ro_community  = 'public',
+  Variant[Undef, String[1], Array[String[1]]]                     $ro_community6 = 'public',
+  Variant[Undef, String[1], Array[String[1]]]                     $rw_community  = undef,
+  Variant[Undef, String[1], Array[String[1]]]                     $rw_community6 = undef,
+  Variant[Array, Stdlib::IP::Address::V4, Stdlib::IP::Address::V4::CIDR] $ro_network    = '127.0.0.1',
```

Comment:
The datatype `Stdlib::IP::Address::V4::CIDR` arrives with stdlib version `4.25.0`.

`metadata.json` update is required to bump stdlib version.

Reply:
Thanks for catching this.